terminal/TerminalServer.cpp

```diff
@@ -153,7 +153,13 @@ void runJumpHost(shared_ptr<ServerClientConnection> serverClientState) {
           if (!serverClientState->readMessage(&message)) {
             break;
           }
-          RawSocketUtils::writeMessage(terminalFd, message);
+          try {
+            RawSocketUtils::writeMessage(terminalFd, message);
+          } catch (const std::runtime_error &ex) {
+            LOG(INFO) << "Terminal session ended";
+            run = false;
+            globalServer->removeClient(serverClientState->getId());
```

Comment:
This line shouldn't be necessary because setting run=false will cause line 316 to fire which does the same thing